### PR TITLE
Robots Header: Stop no-indexing in prod

### PR DESCRIPTION
**Pull request by jarstelfox**
We think the recent proxying change that sends *all* client headers through to the origin may have resulted in the user-agent no longer being `Amazon Cloudfront`. Thus we are now no-indexing all of prod :-(

### CODE REVIEW

**Review by danielbeardsley — DISMISSED** [review]
CR :+1: 

Let's follow this up with a fix. I'll work with @jarstelfox on this.

**Review by danielbeardsley — DISMISSED** [review]
CR :100:

**Comment by danielbeardsley** [comment]
Confirmed, prod currently is 100% no-indexed. So, this will reverse that. It will also index domains we *don't* want to index, but we'll fix that shortly.

File: frontend/helpers/next-helpers.ts [hunk 1]
```diff
@@ -24,16 +24,42 @@ export const withLogging: GetServerSidePropsMiddleware = (next) => {
...
+      context.res?.setHeader('X-Robots-Tag', restrictRobots);
+   }
+}
+
+export function noindexDevDomains(context: ContextType) {
+   if (context.req.headers['user-agent'] !== PROD_USER_AGENT) {
+      // return RestrictRobots.RESTRICT_ALL;
+   }
+
+   return undefined;
```
Comment (danielbeardsley) [inline frontend/helpers/next-helpers.ts:64]:
```suggestion
   return RestricRobot.ALLOW_ALL;
```

**Comment by jarstelfox** [comment]
CC @danny https://vercel.com/guides/are-vercel-preview-deployment-indexed-by-search-engines

It looks like that no index is vercel handling previews themselves

**Comment by danielbeardsley** [comment]
un_dev_block :+1:

File: frontend/helpers/next-helpers.ts (hunk 1 above)
Comment (danielbeardsley) [reply frontend/helpers/next-helpers.ts:64]: Doesn't that apply to this change regardless of which way this goes (i.e. it needs to line up?)

Comment (jarstelfox) [reply frontend/helpers/next-helpers.ts:64]: Fair point. That's why I started refactoring it. The hope was we could have a more general way of getting the indexing string out of a shared function so both the headers and meta tags can reused the code.

**Comment by danielbeardsley** [comment]
> do you have thoughts on QA?

I think curling localhost will be an effective QA (doesn't have vercel preview headers).

**Comment by jarstelfox** [comment]
```
(base)  JARREDS-XPS-13-9310  developer 10:39:39 (no-index-fix) ~/Documents/react-commerce 
▶   curl -i -s http://localhost:3000/products/spudger | grep -i x-robots
(base)  JARREDS-XPS-13-9310  developer 10:39:57 (no-index-fix) ~/Documents/react-commerce E:1
▶  
```

QA 🌵 No x-robots header is sent in local dev

**Review by danielbeardsley — APPROVED** [review]
CR :+1:

**Comment by davidrans** [comment]
This deployment error seems relevant, though I don't see that error in the diff:

https://vercel.com/ifixit/react-commerce-prod/82zQjjP5QV2XzjV3AawPDUVs1y1Z
> Type error: Cannot find name 'RestricRobot'. Did you mean 'RestrictRobots'?

**Comment by jarstelfox** [comment]
@davidrans It was relevant before my force push to walk back that commit: https://github.com/iFixit/react-commerce/compare/38fa30621927ee8891205a3150a6c03cfe33f937..ececf31978e8219f52da30608bb23e336278a1ac

**Pull request merged**